tests/agent_server/test_resource_sets.py

```diff
@@ -93,3 +93,723 @@ async def test_resource_sets_via_put_version(server, client, environment, client
+    assert resource_list[3].attributes == {
+        "key": "key4",
+        "value": "4",
+        "purged": False,
+        "requires": ["test::Resource[agent1,key=key3]"],
+        "send_event": False,
+    }
+    assert resource_list[4].attributes == {"key": "key5", "value": "5", "purged": False, "requires": [], "send_event": False}
+    assert resource_list[5].attributes == {"key": "key6", "value": "6", "purged": False, "requires": [], "send_event": False}
+    assert resource_list[6].attributes == {"key": "key7", "value": "700", "purged": False, "requires": [], "send_event": False}
+    assert resource_sets_from_db == {
+        "test::Resource[agent1,key=key1]": "set-a",
+        "test::Resource[agent1,key=key2]": "set-a",
+        "test::Resource[agent1,key=key3]": "set-b",
+        "test::Resource[agent1,key=key4]": "set-b",
+        "test::Resource[agent1,key=key5]": None,
+        "test::Resource[agent1,key=key6]": None,
+        "test::Resource[agent1,key=key7]": None,
+    }
+
+
+async def test_put_partial_validation_error(server, client, environment, clienthelper):
+    """
+    A partial compile can update multiple resource sets at the same time.
+    """
+    version = await clienthelper.get_version()
+    resources = [
+        {
+            "key": "key1",
+            "value": "value1",
+            "id": "test::Resource[agent1,key=key1],v=%d" % version,
+            "send_event": False,
+            "purged": False,
+            "requires": [],
+        },
+        {
+            "key": "key2",
+            "value": "value2",
+            "id": "test::Resource[agent1,key=key2],v=%d" % version,
+            "send_event": False,
+            "purged": False,
+            "requires": [],
+        },
+    ]
+    resource_sets = {
+        "test::Resource[agent1,key=key1]": "set-a",
+        "test::Resource[agent1,key=key2]": "set-b",
+    }
+
+    result = await client.put_version(
+        tid=environment,
+        version=version,
+        resources=resources,
+        resource_state={},
+        unknowns=[],
+        version_info={},
+        compiler_version=get_compiler_version(),
+        resource_sets=resource_sets,
+    )
+    assert result.code == 200
+    version = await clienthelper.get_version()
+    resources_partial = ["key1", "key2]"]
+
+    result = await client.put_partial(
+        tid=environment,
+        version=version,
+        resources=resources_partial,
+        resource_state={},
+        unknowns=[],
+        version_info=None,
+        compiler_version=get_compiler_version(),
+        resource_sets={
+            "test::Resource[agent1,key=key1]": "set-a",
+            "test::Resource[agent1,key=key2]": "set-b",
+        },
+    )
+
+    assert result.code == 400
+    assert result.result["message"] == (
+        "Invalid request: Type validation failed for resources in put_"
+        "patrial.excepted an argument of type List[Dict[str, Any] but "
+        "received ['key1', 'key2]']"
+    )
```

Comment:
Mentioning the used API endpoint is not very useful here. The user knows which endpoint he/she was invoking.
```suggestion
        "Invalid request: Type validation failed for resources argument. Expected an argument of type List[Dict[str, Any]] but "
        "received ['key1', 'key2]'"
    )
```